From 70c79c05236bac0dd41d104fb4a70a68a0450b75 Mon Sep 17 00:00:00 2001
From: Jon Lin <jon.lin@rock-chips.com>
Date: Sun, 14 Jun 2020 19:21:44 +0800
Subject: [PATCH] drivers: rkflash: Add cast for function min input

warning: comparison of distinct pointer types lacks a cast
error, forbidden warning:kernel.h:733
  (void) (&_min1 == &_min2);  \

Change-Id: I2311d0bbb32be9b44cc262808d9e2df12b9627b9
Signed-off-by: Jon Lin <jon.lin@rock-chips.com>
---
 drivers/rkflash/sfc_nand_mtd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/rkflash/sfc_nand_mtd.c b/drivers/rkflash/sfc_nand_mtd.c
index d22edb86b736..c62335242e48 100644
--- a/drivers/rkflash/sfc_nand_mtd.c
+++ b/drivers/rkflash/sfc_nand_mtd.c
@@ -46,7 +46,7 @@ static int sfc_nand_write_mtd(struct mtd_info *mtd, loff_t to,
 	while (remaining) {
 		memcpy(p_dev->dma_buf, data, mtd->writesize);
 		memset(p_dev->dma_buf + mtd->writesize, 0xff, mtd->oobsize);
-		oob_real = min(oob_left, mtd->oobsize - ops->ooboffs);
+		oob_real = min((u32)oob_left, mtd->oobsize - ops->ooboffs);
 		if (oob_real)
 			memcpy(p_dev->dma_buf + mtd->writesize + ops->ooboffs, oob_buf, oob_real);
 		ret = sfc_nand_prog_page_raw(0, to >> mtd->writesize_shift,
@@ -116,7 +116,7 @@ static int sfc_nand_read_mtd(struct mtd_info *mtd, loff_t from,
 			real_size = mtd->writesize - off;
 
 		memcpy(data, p_dev->dma_buf + off, real_size);
-		oob_real = min(oob_left, mtd->oobsize - ops->ooboffs);
+		oob_real = min((u32)oob_left, mtd->oobsize - ops->ooboffs);
 		if (oob_real)
 			memcpy(oob_buf, p_dev->dma_buf + mtd->writesize + ops->ooboffs,
 			       oob_real);
-- 
2.35.3

